contracts/InterchainTokenService.sol

```diff
@@ -893,20 +855,19 @@ contract InterchainTokenService is
         string memory originalSourceChain = sourceChain;
 
         // Unwrap ITS message if coming from ITS hub
-        if (messageType == MESSAGE_TYPE_RECEIVE_FROM_HUB) {
-            if (keccak256(abi.encodePacked(sourceChain)) != ITS_HUB_CHAIN_NAME_HASH) revert UntrustedChain();
+        if (messageType != MESSAGE_TYPE_RECEIVE_FROM_HUB) {
+            revert InvalidMessageType(messageType);
+        }
 
-            (, originalSourceChain, payload) = abi.decode(payload, (uint256, string, bytes));
+        if (keccak256(bytes(sourceChain)) != ITS_HUB_CHAIN_NAME_HASH) revert UntrustedChain();
```

Comment:
won't the `onlyRemoteService` modifier check this already?

Reply:
nice catch.